### DISCUSSION TITLE
Something is wrong; your library is not initialized

### DISCUSSION THREAD

**Discussion by myshevchuk** [emacs-citar/citar#270] [Q&A] [answered]
I've been experiencing the `Something is wrong; your library is not initialized.` error with Bibtex-actions since a week or two. And the Embark Bibtex-actions commands do not appear upon `embark-act` in the completion buffer, so only the default action is possible with `bibtex-actions-open-entry`, which takes me to the BibTeX entry, which is something I rarely want to actually do.

Not sure if it's worth opening a bug report because I have a feeling it's Doom-related.

**Comment by bdarcus** [comment, marked as answer]
I saw that today too.

And I do know there's a doom embark bug that I think is currently fixed in a PR
for that module, but not yet merged.

I'll explore over the weekend if I have time.

**Comment by bdarcus** [comment]
@myshevchuk In the spirit of not opening an issue that may not be needed ....

In working on related issues, I've gone down a rabbit hole, which touches on ORB.

https://github.com/bdarcus/bibtex-actions/pull/281

Basically what I'm doing there is changing the way bibtex-actions works, so caching is decoupled from formatting.

So on that branch, `bibtex-actions-select-refs` now returns more structured data, rather than a simple list of keys.

```elisp
ELISP> (bibtex-actions-select-keys)
(("agnew2003"
  ("title" . "From Megalopolis to Global City-Region?")
  ("date" . "2003")
  ("author" . "Agnew, John")
  ("=type=" . "article")
  ("=key=" . "agnew2003"))
 ("huck1972"
  ("title" . "The Orders")
  ("date" . "1972")
  ("author" . "Huck, Susan L.M.")
  ("=type=" . "article")
  ("=key=" . "huck1972")))
```

But I earlier added `bibtex-actions-file-open-note-function` with the idea one could specify `orb-edit-note` there.

That function only takes a key as input, and so has to get that data from somewhere else; notably I guess the bibtex-completion cache (here I had a lookup function that was causing this problem that looked in the bibtex-actions caches via `bibtex-actions--get-candidates`).

I guess I'm wondering: do you think it's feasible to add an optional entry parameter on `orb-edit-note` for such structured data, so a user could still use that function here without getting an error?

E.g.:

```elisp
(defun orb-edit-note (citekey &optional _entry)
```

**Reply by myshevchuk** [reply]
@bdarcus Yes, it's absolutely possible. It's also possible for ORB to use the provided entry instead of looking it up. Please issue a feature request for record.
